nixos/modules/services/backup/bupstash.nix

```diff
@@ -0,0 +1,378 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib)
+    mapAttrsToList
+    nameValuePair
+    mkEnableOption
+    mkIf
+    mkMerge
+    mkOption
+    mkPackageOption
+    optional
+    optionalString
+    ;
+
+  inherit (lib.types)
+    attrsOf
+    bool
+    either
+    lines
+    listOf
+    nullOr
+    passwdEntry
+    path
+    str
+    submodule
+    ;
+
+  mkJobTarget = name: _: {
+    name = "bupstash-${name}";
+    value = {
+      description = "Bupstash group job for ${name}.";
+      unitConfig = {
+        StopWhenUnneeded = true;
+      };
+    };
+  };
+
+  mkJobTimer = name: job: {
+    name = "bupstash-${name}";
+    value = {
+      wantedBy = [ "timers.target" ];
+      timerConfig = {
+        Unit = "bupstash-${name}.target";
+        OnCalendar = job.startAt;
+      };
+    };
+  };
+
+  mkJobUnits =
+    name: job:
+    [
+      {
+        name = "bupstash_prehook-${name}";
```

Comment:
The idea is to prevent name clashing if someone has the brilliant idea to call their backup `prehook`, but as we only have a target and a timer with the short name it should be fine to use a dash

Reply:
I replaced the name by `bupstash-${name}-prehook`, which allow filtering directly on `bupstash-${name}`